src/astral/page-script.js

```diff
@@ -205,19 +205,29 @@ async function updateHpBar({characterName, hp, maxHp, tempHp}) {
             console.warn(`Couldn't update the character hp due to the following reason: No character found with name ${characterName}`)
             return
         }
+        const characterData = await getCharacterData(character);
```

Comment:
That function doesn't do any checks, you'd need to catch it in case it errors out, and handle the fail case so it doesn't prevent the rest from happening.

Reply:
Well without character data it would be pointless to continue either way since we don't have the data structures and objects to update the char sheet. So I would say it would be better to crash here and the error would be caught in the catch below and logged.

Reply:
shouldn't it fallback on what you had before which didn't use the character data? 
```
                    attributeBarMax: maxHp + tempHp,
                    attributeBarCurrent: hp + tempHp,
```

Reply:
The one previous isn't use anymore, I think those attributes are deprecated.

Reply:
ah ok, thanks!